katsdpcontroller/scheduler.py

```diff
@@ -944,20 +992,32 @@ async def __call__(self, repo: str, tag: str) -> str:
             ssl_context = ssl.create_default_context(cafile=cafile)
         else:
             ssl_context = None
-        async with aiohttp.ClientSession(
-                headers={'Accept': 'application/vnd.docker.distribution.manifest.v2+json'},
-                auth=self._auth) as session:
+        async with aiohttp.ClientSession() as session:
             try:
-                # Use a lowish timeout, so that we don't wedge the entire launch if
-                # there is a connection problem.
-                async with session.head(url, timeout=15, ssl_context=ssl_context) as response:
-                    response.raise_for_status()
-                    digest = response.headers['Docker-Content-Digest']
-            except (aiohttp.client.ClientError, asyncio.TimeoutError) as error:
-                raise ImageError('Failed to get digest from {}: {}'.format(url, error)) \
-                    from error
-            except KeyError:
-                raise ImageError('Docker-Content-Digest header not found for {}'.format(url))
+                auth_header = auth.encode() if auth else None
+                digest = await self._get_digest(session, url, ssl_context, auth_header)
+            except ImageError as error:
+                cause = error.__cause__
```

Comment:
This is a little gem. I was not aware of context coming through with exception like this.